leaf-ui/rappid-extensions/BackboneModels.js

```diff
@@ -1 +1,455 @@
-/** This file contains backbone model representations of the original model objects - WIP */
\ No newline at end of file
+/** This file contains backbone model representations of the original model objects - WIP */
+
+// FuncSegment Backbone Model from FuncSegment class in modelObjects.js
+var FunctionSegmentBBM = Backbone.Model.extend({
+    idAttribute: "uid",
+
+    // Initialize function - save all of the input parameters into one 'option' input
+    // And then use 'option' when intitializing the function 
+    initialize: function (options) {
+        this.type = options.type;
+        this.refEvaluationValue = options.refEvaluationValue;
+        this.startTP = options.startTP;
+        this.stopTP = options.stopTP;
+    },
+});
+
+// RepFuncSegment Backbone Model from RepFuncSegment class in modelObjects.js
+var RepFuncSegmentBBM = Backbone.Model.extend({
+    idAttribute: "uid",
+
+    initialize: function(options) { 
+        // this functionSegList is an array of all of the FunctionSegmentBBMs in the RepFuncSegmentBBM
+        this.functionSegList = options.functionSegList; 
+        this.repNum = options.repNum; 
+        this.absRepTime = options.absRepTime; 
+    }
+
+});   
+
+
+
+// Backbone Model of Evolving Function Class on modelObjects.js
+var EvolvingFunctionBBM = Backbone.Model.extend({
+    idAttribute: "uid",
+
+    initialize: function(options) { 
+        //this.intentionID = options.intentionID; 
+        this.nodeID = options.nodeID;  
+        // Not Sure if calling the other models inside this one is done correctly
+        var FuncSegment = new FuncSegmentModel();
+        var FuncSegmentOptions = ; // Have to eventually add all of the FuncSegment parameters here

```

Comment:
Can remove this

Reply:
Options can be passed directly into the model creation. For example: new FuncSegmentModel({nodeID: 0000, otherVariable1: "exampleVar1, othervariable2: "exampleVar2"})